File: backend/app/schemas/service_record.py
"""Service history schemas — mirror the `service_records` table."""

from app.schemas.user import CamelModel


class ServiceRecordOut(CamelModel):
    id: str
    service_date: str
    contractor_name: str | None = None
    work_performed: str
    cost: int | None = None
    linked_document_id: str | None = None
    notes: str | None = None


class ServiceRecordCreate(CamelModel):
    service_date: str
    contractor_name: str | None = None
    work_performed: str
    cost: int | None = None
    linked_document_id: str | None = None
    notes: str | None = None

File: frontend/lib/homeowner/types/service-record.ts
/** Service history log — mirror backend/app/homeowner/schemas/service_record.py
 * and the `service_records` table.
 */

export interface ServiceRecord {
  id: string;
  serviceDate: string;
  contractorName: string | null;
  workPerformed: string;
  cost: number | null;
  linkedDocumentId: string | null;
  notes: string | null;
}

export interface ServiceRecordCreate {
  serviceDate: string;
  contractorName?: string | null;
  workPerformed: string;
  cost?: number | null;
  linkedDocumentId?: string | null;
  notes?: string | null;
}
